libr/core/cmd_print.c

```diff
@@ -6541,7 +6541,7 @@ static int cmd_print(void *data, const char *input) {
 		r_cons_break_push (NULL, NULL);
 		switch (input[1]) {
 		case 'j': // "pxj"
-			r_print_jsondump (core->print, core->block, core->blocksize, 8);
+			r_print_jsondump (core->print, core->block, core->num->value, 8);
```

Comment:
Can you test this patch? See other cases how it takes the data.
```suggestion
			r_print_jsondump (core->print, core->block, len, 8);
```

also can you add a test to ensure the output is correct for positive and negative offsets, as well as 0

Reply:
In the tests try also bigger output than the blocksize. Something like:

```
pxj 10
pxj -10
b 16
pxj
pxj 32
pxj-32
```

Reply:
```
b 16
pxj
pxj 32
pxj-32
```
The following tests produced a wrong output from the negative value, it cames 16 bytes instead 32

Reply:
Yep, i can confirm this issue too. But i think it's fixed if you pass buf and len.

Please create a new test by adding those lines in `test/db/cmd/cmd_print` and run it like this `r2r -i test/db/cmd/cmd_print`.

Reply:
ping?